### PR TITLE
feat: adding support for element.matches

### PR DESCRIPTION
**Pull request by tedconn**
With this PR we add support for a shadow piercing [`Element.matches()`](https://developer.mozilla.org/en-US/docs/Web/API/Element/matches).

### CODE REVIEW

**Comment by tedconn** [comment]
Actually I just discovered a bug in this PR. Writing tests for it. Please don't merge.

**Review by caridy — APPROVED** [review]

File: test/matches.spec.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import simpleShadow1 from './fixtures/simple1/shadow.html'
+import assert from 'assert'
+
+describe(('element matches'), () => {
+  const matchingSelector = '.container .component span'
+
+  it('light DOM - matches', () => {
+    withDom(simpleLight1, context => {
+      const element = context.querySelector(matchingSelector)
+      assert(element.matches(matchingSelector) === true)
```
Comment (nolanlawson) [inline test/matches.spec.js:20]: `assert.deepStrictEqual()` is preferable here (better debug message if it fails)

File: types/kagekiri.d.ts [hunk 2]
```diff
@@ -12,3 +12,4 @@ export function getElementsByTagName(tagName: string, context?: Node): Element[]
 export function getElementsByTagNameNS(namespaceURI: string, localName: string, context?: Node): Element[];
 export function getElementById(id: string, context?: DocumentOrShadowRoot): Element | null;
 export function getElementsByName(name: string, context?: DocumentOrShadowRoot): Element[];
+export function matches(selector: string, context?: Node): boolean;
```
Comment (nolanlawson) [inline types/kagekiri.d.ts:15]: `context` is not optional here, correct? I think we should remove the `?`.

**Review by nolanlawson — CHANGES_REQUESTED** [review]

**Comment by nolanlawson** [comment]
Small nits, otherwise LGTM!

File: types/kagekiri.d.ts (hunk 2 above)
Comment (tedconn) [reply types/kagekiri.d.ts:15]: You're right! It kept bugging me that something was off.

File: test/matches.spec.js (hunk 1 above)
Comment (tedconn) [reply test/matches.spec.js:20]: Yeah, I changed to `strictEqual`

**Review by nolanlawson — APPROVED** [review]

**Pull request merged**